### DISCUSSION TITLE
Confusion for private keys

### DISCUSSION THREAD

**Discussion by HakooDesai** [passkeydeveloper/community#9] [General Q&A] [answered]
In general, I understood that during registration ceremony, Authenticator generates the Pub/Private key pair, stores a private key and provides public key in response. Now my question is, what happens if Authenticator doesn't store the Private key, i.e. **supports residential keys = false?**

Also, does Authenticator generates the pair every time it registers with new website? 

Can I use same authenticator for same website for multiple usernames?

**Comment by timcappalli** [comment, marked as answer]
> what happens if Authenticator doesn't store the Private key, i.e. supports residential keys = false?

If you request a passkey correctly in the create call, the authenticator will only create a passkey (you can confirm in the credProps response).

> Also, does Authenticator generates the pair every time it registers with new website?

Yes, passkeys are unique per RP ID, by design.

> Can I use same authenticator for same website for multiple usernames?

Yes. Each passkey is in an authenticator is unique per user account + RP ID combo.
